lib/teleterm/clusters/dbcmd_cli_command_provider.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clusters
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/gravitational/teleport/lib/client/db/dbcmd"
+	apiuri "github.com/gravitational/teleport/lib/teleterm/api/uri"
+	"github.com/gravitational/teleport/lib/teleterm/gateway"
+	"github.com/gravitational/teleport/lib/tlsca"
+
+	"github.com/gravitational/trace"
+)
+
+// DbcmdCLICommandProvider provides CLI commands for database gateways. It needs Storage to read
+// fresh profile state from the disk.
+type DbcmdCLICommandProvider struct {
+	storage *Storage
+}
+
+func NewDbcmdCLICommandProvider(storage *Storage) DbcmdCLICommandProvider {
+	return DbcmdCLICommandProvider{
+		storage: storage,
+	}
+}
+
+func (d DbcmdCLICommandProvider) GetCommand(gateway *gateway.Gateway) (string, error) {
```

Comment:
@rosstimothy I'd love to add more tests here, I shunned doing that because I don't have much experience writing tests in Go.

Do you have any tips on how I could deal with `dbcmd.NewCmdBuilder` in tests? Ideally I'd like to just verify that `GetCommand` calls `dbcmd.NewCmdBuilder` with correct arguments. I don't want to actually call it in tests since by `dbcmd` calls stuff like `exec.LookPath`. 

dbcmd tests deal with that by changing the private `exe` field of `CLICommandBuilder` to a mock implementation which I cannot do.

There are two options which come to my mind:

1. Refactor the whole `dmcmd.NewCmdBuilder` so that `exe` can be provided from the outside, perhaps through a `ConnectCommandFunc`.
2. Refactor `DbcmdCLICommandProvider` so that the struct has a field which holds a function of the same signature as `dbcmd.NewCmdBuilder`. In tests that would be replaced with a mock implementation which just verifies (somehow) that the correct args were passed.

The TDD purists in me tells me that the second option would be more correct, but the pragmatist would probably prefer the first one.

Reply:
@rosstimothy I thought about it some more and I feel like adding another interface wouldn't help much. If I added `DBCommandBuilder` then I could test that `DbcmdCLICommandProvider` passes the right arguments to `DBCommandBuilder`.

But the actual value behind writing the tests would be in making sure that `dbcmd.NewCmdBuilder` receives correct arguments. Which means I'd need to mock something that closely resembles `dbcmd.NewCmdBuilder` interface but I feel like the interface of `dbcmd.NewCmdBuilder` is quite hard to mock/test, at least with my limited Go knowledge.

---

To help with the absence of tests at least a little bit, I reduced the surface area of `DbcmdCLICommandProvider` by extracting methods that are not essential to building a command. Would that be fine for now? I'd like to merge this feature before we cut the RC v10 version.